app/src/main/java/protect/card_locker/ShortcutHelper.java

```diff
@@ -155,4 +155,26 @@ Bitmap createAdaptiveBitmap(@NotNull Bitmap in, int paddingColor) {
                 .setIntent(intent)
                 .setIcon(icon);
     }
+    static ShortcutInfoCompat.Builder createGroupShortcutBuilder(Context context, Group group) {
+        Intent intent = new Intent(context, MainActivity.class);
+        intent.setAction(Intent.ACTION_MAIN);
+        // Prevent instances of the view activity from piling up; if one exists let this
+        // one replace it.
+        intent.setFlags(intent.getFlags() | Intent.FLAG_ACTIVITY_SINGLE_TOP);
+        final Bundle bundle = new Bundle();
+        // Done like with the cards but with group._id (a string)
+        bundle.putString("id", group._id);
+        bundle.putBoolean("view", true);
+        intent.putExtras(bundle);
+
+        Bitmap iconBitmap = Utils.generateIcon(context, group, true).getLetterTile();
+
+        IconCompat icon = IconCompat.createWithAdaptiveBitmap(iconBitmap);
+
+        return new ShortcutInfoCompat.Builder(context, group._id)
+                .setShortLabel(group._id)
+                .setLongLabel(group._id)
+                .setIntent(intent)
+                .setIcon(icon);
```

Comment:
Given groups don't inherently have their own icon, why not just use Catima's logo here instead? That way it is also clear from the widget that this won't open a card.

Reply:
That is a good idea, however how could we differentiate between a group widget and the app icon except the title? Would that be enough?

Reply:
I think the title would be sufficient for now